sdks/python/apache_beam/io/gcp/gcsio.py

```diff
@@ -110,6 +110,27 @@ def parse_gcs_path(gcs_path, object_optional=False):
   return match.group(1), match.group(2)
 
 
+def default_gcs_bucket_name(project, region):
+  from hashlib import md5
+  return 'dataflow-staging-%s-%s' % (
+      region, md5(project.encode('utf8')).hexdigest())
+
+
+def get_or_create_default_gcs_bucket(project, region, kms_key=None):
```

Comment:
I realize that there's a check for this in `_create_default_gcs_bucket`, but this method should fail if kms_key is set. Otherwise, it's saying that it's okay to have a default bucket with a KMS key.

(The default bucket should not use a KMS key, and if a KMS key is specified Beam should not use a bucket with some other key (KMS or Google managed).)

Refs:
https://github.com/apache/beam/pull/8135#discussion_r274695249
https://github.com/apache/beam/pull/8830/files

Reply:
okay I've made this not work in that case. WDYT?

Reply:
I still need to review the rest of the code. Can look tomorrow, overloaded today